jujugui/static/gui/src/app/components/user-profile/create-model-button/create-model-button.js

```diff
@@ -0,0 +1,155 @@
+Copyright (C) 2016 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('create-model-button', function() {
+
+  juju.components.CreateModelButton = React.createClass({
+    propTypes: {
+      addNotification: React.PropTypes.func.isRequired,
+      className: React.PropTypes.string,
+      controllerAPI: React.PropTypes.object.isRequired,
+      hideConnectingMask: React.PropTypes.func.isRequired,
+      jem: React.PropTypes.object,
+      showConnectingMask: React.PropTypes.func.isRequired,
+      switchModel: React.PropTypes.func.isRequired,
+      user: React.PropTypes.object
+    },
+
+    getInitialState: function() {
+      return {
+        createNewModelActive: false
+      };
+    },
+
+    getDefaultProps: function() {
+      return {
+        className: ''
+      };
+    },
+
+    /**
+      Fetches the model name from the modelName ref and then creates a model
+      then switches to it.
+      @method createAndSwitch
+      @param {Object} e The event handler from a form submission.
+    */
+    createAndSwitch: function(e) {
+      if (e && e.preventDefault) {
+        e.preventDefault();
+      }
+      // The supplied new model name needs to be valid.
+      const modelName = this.refs.modelName;
+      if (!modelName.validate()) {
+        // Only continue if the validation passes.
+        return;
+      }
+      const props = this.props;
+      // Because this will automatically connect to the model lets show
+      // the connecting mask right now.
+      props.showConnectingMask();
+      props.controllerAPI.createModel(
+        modelName.getValue(),
+        props.user.user,
+        data => {
+          const err = data.err;
+          if (err) {
+            props.addNotification({
+              title: 'Failed to create new Model',
+              message: err,
+              level: 'error'
+            });
+            console.error(err);
+            props.hideConnectingMask(false);
+            return;
+          }
+          props.switchModel(data.uuid, data.name);
+        });
+    },
+
+    /**
+      Depending on the existance of jem this will either switch to a
+      disconnected model or open up the UI to allow the user to create a
+      new model.
+      @method _nextCreateStep
+    */
+    _nextCreateStep: function() {
+      if (this.props.jem) {
```

Comment:
With `jem` gone this will likely have to be flagged off the existence of the app level key `jimmURL` which we should probably set in controllerAPI instance so that you could do `this.props.controllerAPI.get('jimmURL')` here.